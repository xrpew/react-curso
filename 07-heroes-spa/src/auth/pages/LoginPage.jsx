import { useContext } from "react";
import { useNavigate } from "react-router-dom";
import { AuthContext } from "../context/AuthContext";

export const LoginPage = () => {


  const {login} = useContext(AuthContext)
  const navigate = useNavigate();
    
  const onLogin = ()=>{

    const lastPath = localStorage.getItem('lastPath') || '/';

    login('Sergio PÃ©rez');

      navigate( lastPath,{
          replace:true
      });
  }
  

  return (
    <div className="container mt-5">
      <h1>Login</h1>
      <hr />
      <button
      onClick={onLogin}
      className="btn btn-primary">

      Login

      </button>

    </div>
    )
}
