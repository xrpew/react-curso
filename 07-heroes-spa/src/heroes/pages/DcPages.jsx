import { HeroList } from "../components";

export const DcPages = () => {
  return (
    <>
      <h1>DC Comics</h1>
      <hr />

      <HeroList publisher="DC Comics" />
    </>
  );
};
