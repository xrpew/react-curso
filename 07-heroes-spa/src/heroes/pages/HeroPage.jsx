import { useMemo } from 'react';
import { Navigate, useNavigate, useParams } from 'react-router-dom'
import { getHeroeById } from '../helpers';

export const HeroPage = () => {

  const { id } = useParams();
  const navigate = useNavigate();

 const hero = useMemo(()=> getHeroeById(id), [id]);

 const onNavigateBar = ()=>{
  navigate(-1);
}
 
 
if( !hero ){
  return <Navigate to="/marvel" />
}


  return (
    <div className="row mt-5">
      <div className="col-4">
        <img 
          src={`/assets/heroes/${ id }.jpg`} 
          alt={hero.superhero}
          className="animate__animated animate__fadeInLeft img-complete" />
      </div>
      <div className="col-8">
        <h3>{hero.superhero}</h3>
        <ul className="list-group list-group-flush">
          <li className="list-group-item"><b>Alter ego:</b> {hero.alter_ego} </li>
          <li className="list-group-item"><b>Publisher:</b> {hero.publisher} </li>
          <li className="list-group-item"><b>First appaerance:</b> {hero.first_appearance} </li>

        </ul>

        <h5 className="mt-3">Characters</h5>
        <p>{hero.characters}</p>
      </div>

    <button className="btn btn-outline-primary"
    onClick={onNavigateBar}>
      Regresar
    </button>
    </div>
  )
}
