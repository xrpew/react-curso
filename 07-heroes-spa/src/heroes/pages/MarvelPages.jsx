import { HeroList } from "../components";

export const MarvelPages = () => {
  return (
    <>
      <h1>Marvel Comics</h1>
      <hr />

      <HeroList publisher="Marvel Comics" />
    </>
  );
};
