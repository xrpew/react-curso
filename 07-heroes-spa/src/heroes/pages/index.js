export *  from './DcPages'
export * from './MarvelPages'
export * from './SearchPage'
export * from './HeroPage'