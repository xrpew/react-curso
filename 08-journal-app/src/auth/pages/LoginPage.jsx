import { useMemo } from "react";
import { Link as RouterLink } from "react-router-dom";
import { Google } from "@mui/icons-material";
import { Alert, Button, Grid, Link, TextField, Typography } from "@mui/material";
import { AuthLayout } from "../layout/AuthLayout";
import { useForm } from "../../hooks";
import { useDispatch, useSelector } from "react-redux";
import {
  checkingAuthentications,
  startGoogleSingIn,
  startLoginWithEmailPassword,
} from "../../store/auth";



const formData= { 
  email:'',
  password:''
}

export const LoginPage = () => {
  const { status, errorMessage } = useSelector((state) => state.auth);

  const dispatch = useDispatch();



  const { email, password, onInputChange } = useForm( formData );

  const isAuthenticating = useMemo(() => status === "checking", [status]);

  const onSubmit = (event) => {
    event.preventDefault();
    // console.log({email, password});
    // dispatch( checkingAuthentications() );
    dispatch(startLoginWithEmailPassword({ email, password }));
  };
  const onGoogleSingIn = () => {
    dispatch(startGoogleSingIn());
  };

  return (
    <AuthLayout title="Login">
      <form 
      className='animate__animated animate__fadeIn animate__faster'
      onSubmit={onSubmit}>
        <Grid container>
          <Grid item xs={12} sx={{ mt: 2 }}>
            <TextField
              label="correo"
              type="email"
              placeholder="correo@correo.corr"
              fullWidth
              name="email"
              value={email}
              onChange={onInputChange}
            />
          </Grid>
          <Grid item xs={12} sx={{ mt: 2 }}>
            <TextField
              label="contraseña"
              type="password"
              placeholder="contraseña"
              fullWidth
              name="password"
              value={password}
              onChange={onInputChange}
            />
          </Grid>

          <Grid container display={!!errorMessage ? "" : "none"} sx={{mt:2}}>
              <Grid item xs={12} >
                <Alert severity="error">{errorMessage}</Alert>
              </Grid>
            </Grid>

          <Grid container spacing={2} sx={{ mb: 2, mt: 1 }}>
            <Grid item xs={12} sm={6}>
              <Button
                disabled={isAuthenticating}
                type="submit"
                variant="contained"
                fullWidth
              >
                Login
              </Button>
            </Grid>

            

            <Grid item xs={12} sm={6}>
              <Button
                disabled={isAuthenticating}
                onClick={onGoogleSingIn}
                variant="contained"
                fullWidth
              >
                <Google />
                <Typography sx={{ ml: 1 }}>Google</Typography>
              </Button>
            </Grid>
          </Grid>

          <Grid container direction="row" justifyContent="end">
            <Link component={RouterLink} color="inherit" to="/auth/register">
              Crear una cuenta
            </Link>
          </Grid>
        </Grid>
      </form>
    </AuthLayout>
  );
};
