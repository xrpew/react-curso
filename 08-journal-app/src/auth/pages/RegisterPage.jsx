import { useMemo, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link as RouterLink } from 'react-router-dom';
import { Alert, Button, Grid, Link, TextField, Typography } from '@mui/material';
import { AuthLayout } from '../layout/AuthLayout';
import { useForm } from '../../hooks';
import { startCreatingUserWithEmailPassword } from '../../store/auth';

const formData = {
  email:'',
  password:'',
  displayName:''
};

const formValidations = {
  email:[ (value)=> value.includes('@') , 'Debes ingresar un correo válido.' ],
  password:[ (value)=> value.length >=6 , 'El password debe tener mas de 6 letras.' ],
  displayName:[ (value)=> value.length >=1 , 'El nombre es obligatorio.' ],
}


export const RegisterPage = () => {

  const {status, errorMessage} = useSelector(state=> state.auth)
  const isChekingAuthentication = useMemo(()=> status === 'cheking', [status]);

  const dispatch = useDispatch();

  const [formSubmited, setFormSubmited] = useState(false);


  const { formState,   displayName,      email,      password, onInputChange,  
          isFormValid, displayNameValid, emailValid, passwordValid } 
  = useForm( formData, formValidations );

    


  const onSubmit = (event) => {
    event.preventDefault();
    setFormSubmited(true);
    if(!isFormValid) return
    dispatch( startCreatingUserWithEmailPassword(formState));
  };

  return (
    <AuthLayout title="Crear Cuenta">
        <h1>FormValid: { isFormValid ? 'valido':'incorrecto'}</h1>
      <form 
      className='animate__animated animate__fadeIn animate__faster'
      onSubmit={onSubmit}>
        <Grid container>
          <Grid item xs={12} sx={{ mt: 2 }}>
            <TextField
              label="Nombre Completo"
              type="text"
              placeholder="Nombre completo"
              fullWidth
              name="displayName"
              value={displayName}
              onChange={onInputChange}
              error= {!!displayNameValid && formSubmited}
              helperText={displayNameValid}
            />
          </Grid>
          <Grid item xs={12} sx={{ mt: 2 }}>
            <TextField
              label="correo electronico"
              type="email"
              placeholder="correo@electronico.com"
              fullWidth
              name="email"
              value={email}
              onChange={onInputChange}
              error= {!!emailValid && formSubmited}
              helperText={emailValid}
            />
          </Grid>
          <Grid item xs={12} sx={{ mt: 2 }}>
            <TextField
              label="contraseña"
              type="password"
              placeholder="contraseña"
              fullWidth
              name="password"
              value={password}
              onChange={onInputChange}
              error= {!!passwordValid && formSubmited}
              helperText={passwordValid}
            />
          </Grid>

          <Grid container spacing={2} sx={{ mb: 2, mt: 1 }}>
          <Grid item xs={12} display={!!errorMessage ? '': 'none'}>
              <Alert severity='error'>{ errorMessage}</Alert>
            </Grid>
            <Grid item xs={12}>
              <Button  disabled={ isChekingAuthentication } type="submit" variant="contained" fullWidth>
                Crear cuenta
              </Button>
            </Grid>
          </Grid>

          <Grid container direction="row" justifyContent="end">
            <Typography sx={{ mr: 1 }}>¿Ya tienes cuenta?</Typography>
            <Link component={RouterLink} color="inherit" to="/auth/login">
              Ingresar
            </Link>
          </Grid>
        </Grid>
      </form>
    </AuthLayout>
  );
};
