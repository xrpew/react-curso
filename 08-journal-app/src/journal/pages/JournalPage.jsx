import { useDispatch, useSelector } from "react-redux";
import { AddOutlined } from "@mui/icons-material";
import { IconButton, Typography } from "@mui/material";
import { startNewNote } from "../../store/journal/thunks";
import { JournalLayout } from "../layout/JournalLayout";
import { NoteView, NothingSelectedView } from "../views";

export const JournalPage = () => {
  const { isSaving, active } = useSelector( state=> state.journal);
  const dispatch = useDispatch();
  const onClickNewNote = () => {
    dispatch( startNewNote() );
  };

  return (
    <JournalLayout>
      
{
    (!!active) ?  <NoteView/> : <NothingSelectedView/>
}

      <IconButton
      onClick={onClickNewNote}
        size="large"
        disabled={ isSaving }
        sx={{
          color: "white",
          backgroundColor: "error.main",
          ":hover": { backgroundColor: "error.main", opacity: 0.9 },
          position: "fixed",
          right: 50,
          bottom: 50,
        }}
      >
        <AddOutlined sx={{ fontSize: 30 }} />
      </IconButton>
    </JournalLayout>
  );
};
