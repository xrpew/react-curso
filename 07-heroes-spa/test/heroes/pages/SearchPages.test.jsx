import { fireEvent, render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import { SearchPage } from "../../../src/heroes/pages/SearchPage"



const mockUseNavigate = jest.fn();

jest.mock('react-router-dom', ()=>({
    ...jest.requireActual('react-router-dom'),
    useNavigate: ()=> mockUseNavigate
}))




describe('pruebas en search pages', () => {


    beforeEach(()=>jest.clearAllMocks() );
  test('should mostrarse correctamente con valores por defecto', () => { 
    
    const { container } =render(
        <MemoryRouter>
            <SearchPage/>
        </MemoryRouter>
    )

        expect(container).toMatchSnapshot();

   })
   test('debe de mostrar a batman y el input con el valor de queri string', () => { 
    
    render(
        <MemoryRouter initialEntries={['/search?q=batman']}>
            <SearchPage/>
        </MemoryRouter>
    )

        const input = screen.getByRole('textbox');
        expect(input.value).toBe('batman')
        const img = screen.getByRole('img')
        expect( img.src ).toContain('/assets/heroes/dc-batman.jpg')
              
   })

   test('debe de mostrar un error si no hay hero ( batman123 )',()=>{
    render(
        <MemoryRouter initialEntries={['/search?q=batman123']}>
            <SearchPage/>
        </MemoryRouter>
    )

    expect(screen.getByText("There's no result with")).toBeTruthy()    

   });
   test('debe de llamar al navigate a la pantalla nueva',()=>{

    const inputValue = 'superman'
        render(
        <MemoryRouter initialEntries={['/search']}>
            <SearchPage/>
        </MemoryRouter>
        )
            const input = screen.getByRole('textbox');
            fireEvent.change(input, {target:{name:'searchText', value: inputValue}})

        const form = screen.getByRole('form')
        fireEvent.submit(form)

        expect( mockUseNavigate ).toHaveBeenCalledWith(`?q=${inputValue}`)
   })

})
