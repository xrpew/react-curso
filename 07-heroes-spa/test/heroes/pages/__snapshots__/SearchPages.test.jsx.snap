// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en search pages should mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    SearchPage
  </h1>
  <div
    class="row"
  >
    <hr />
    <div
      class="col-5"
    >
      <h4>
        Searching..
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="seach a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-2"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        aria-label="alert-danger"
        class="alert alert-primary"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;
